baggage/baggage.go

```diff
@@ -550,3 +530,125 @@ func (b Baggage) String() string {
 	}
 	return strings.Join(members, listDelimiter)
 }
+
+// parsePropertyInternal attempts to decode a Property from the passed string. It follows the spec at
+// https://www.w3.org/TR/baggage/#definition.
+func parsePropertyInternal(s string) (p Property, ok bool) {
+	// Attempting to parse the key.
+	index := skipSpace(s, 0)
+
+	keyStart := index
+	keyEnd := index
+	for _, c := range s[keyStart:] {
+		if !validateKeyChar(c) {
+			break
+		}
+		keyEnd++
+	}
+
+	if keyStart == keyEnd {
+		// Empty string after skipping whitespaces.
+		return
+	}
+
+	index = skipSpace(s, keyEnd)
+
+	if index == len(s) {
+		// There is only a key (no value).
+		ok = true
+		p.key = s[keyStart:keyEnd]
+		return
+	}
+
+	if s[index] != keyValueDelimiter[0] {
```

Comment:
because of this one: `strings.Cut(keyValue, keyValueDelimiter)`. We would need to create here a string every time it gets called. Not entirely sure about all the compiler optimizations, but assuming there aren't any. It is cheaper to keep this as a string and use index0 than to create a string every time, wdyt?